### PR TITLE
refactoring visual diff tests

### PR DESCRIPTION
**Pull request by dmackenz**
Before the HTML files were too large and it was causing some of the test results to render as blank. By splitting them into smaller HTML files this fixed the issue.

### CODE REVIEW

**Comment by dmackenz** [comment]
@AlexBedley @rowansilver The initial push to this PR should've definitely been a failed build due to the goldens changing. Why are the visual diff tests not running?

File: test/perceptual/d2l-grade-result-read-only.visual-diff.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable prefer-arrow-callback */
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+const testDiff = require('./utils.js');
+
+describe('read only visual diff tests', () => {
+
+	const visualDiff = new VisualDiff('read-only', __dirname);
```
Comment (dbatiste) [inline test/perceptual/d2l-grade-result-read-only.visual-diff.js]: A more unique name should be used.  This value is used to create the folder in the S3 bucket for the reports.  Generally, I recommend using a name that maps nicely, or is exactly the name of the component.  In core we use the name of the component, without the `d2l-`.  If more than one visual-diff uses the same name, I think we will have problems.

File: test/perceptual/utils.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+async function focusGradesButton(page, id) {
+	await page.evaluate((id) => {
```
Comment (dbatiste) [inline test/perceptual/utils.js]: Tip: there is a short-cut for executing code in the page that requires selecting an element.  Just avoids the id-passing, but you'd still need to poke at shadow dom.
```
await page.$eval(`${id} .d2l-labs-d2l-grade-result-presentational`, (elem) => {
    elem.shadowRoot...;
    ...
});
```
🤷‍♂

File: test/perceptual/d2l-grade-result-read-only.visual-diff.js (hunk 1 above)
Comment (dbatiste) [reply test/perceptual/d2l-grade-result-read-only.visual-diff.js]: Actually, looks like each repo will get it's own folder, so it's _probably_ ok.

Comment (dmackenz) [reply test/perceptual/d2l-grade-result-read-only.visual-diff.js]: I will make this change

**Pull request merged**